src/emulator.ts

```diff
@@ -592,4 +600,64 @@ export class EmacsEmulator implements IEmacsController, vscode.Disposable {
       }
     });
   }
+
+  public saveRegister(arg: string): void {
+    const selectionsAfterRectDisabled =
+      this.inRectMarkMode &&
+      this.nativeSelections.map((selection) => {
+        const newLine = selection.active.line;
+        const newChar = Math.min(selection.active.character, selection.anchor.character);
+        return new vscode.Selection(newLine, newChar, newLine, newChar);
+      });
+    const selections = getNonEmptySelections(this.textEditor);
+    if (selectionsAfterRectDisabled) {
+      this.textEditor.selections = selectionsAfterRectDisabled;
+    }
+    // selections is now a list of non empty selections, iterate through them and
+    // build a single variable combinedtext
+    let i = 0;
+    let combinedtext = "";
+    while (i < selections.length) {
+      combinedtext = combinedtext + this.textEditor.document.getText(selections[i]);
+      i++;
+    }
+
+    const register_string = arg;
+    if (register_string == null) {
+      return;
+    }
+
+    const msg = "Copying selection to register " + arg;
+    MessageManager.showMessage(msg);
```

Comment:
Looks like the orignal Emacs doesn't show this message, so we shouldn't do it too?

Reply:
Agreed. I've removed both the copying to register message and the inserting from register message.